Assets/ArenaSDK/SharedScripts/ArenaAgent.cs

```diff
@@ -826,23 +832,28 @@ private void
                 DestroyCollider(ID);
                 ApplyTeamMaterial(ID);
                 Utils.TransparentObject(ID);
-                Utils.TextAllTextMeshesInChild(ID, gameObject.GetComponent<ArenaNode>().GetLogTag());
+                Utils.TextAllTextMeshesInChild(ID, GetArenaNode().GetLogTag());
             } else {
                 Debug.Log(
                     "No ID in this agent, this may cause the agent teammates hard to identidy each other, add the ID prefab in your agent.");
             }
+
+            Debug.LogWarning(
```

Comment:
As for the binary encoder, let's first do the simplest thing, which is encoding social tree coordinates in the binary encoder. I have left an interface here. Can you take over from here? @wandrzej

Reply:
After this, I will add actions that send messages via the binary encoder.

Reply:
I am working on this now.